### ISSUE TITLE
Proposal: Command line reporter for Mochawesome merged results

### ISSUE THREAD

**Issue by brettz9** [cypress-io/cypress#6585]
### Current behavior:

1. Use the behavior described at https://docs.cypress.io/guides/tooling/reporters.html#Merging-reports-across-spec-files for merging files
2. Run the Cypress tests (perhaps with a lot of noise added between tests, whether due to the Cypress logging level or other parallel processes added into the testing).
3. To get back to just the test result data, I must either scroll through a lot of noise on the command line, or look at the built HTML report.

### Desired behavior:

After steps 1 and 2 above, in place of step 3, I'd like to have the verbose CLI testing process be concluded with a report (whether as a brief summary or which shows individual failing tests with no noise interspersed between the tests). I'd also like to be able to look up the last test results at a future time (similar to `nyc report` for coverage).

In other words, I'd like a CLI reporter which works on the Mochawesome-generated report (I'm not actually tied to Mochawesome specifically, but it seems their merge tool may be unique). Though I like the ability to get an HTML report, I sometimes just want to use a CLI.

FWIW, here is some code I wrote to do this, and it seems to work well across all default mocha reporters (besides "html" which is not intended for CLI use). With many other projects on my plate, I feel too busy at the moment to prepare a repo for it myself, but I thought Cypress might be interested in having such a solution.

```js
// See https://github.com/jsdoc/jsdoc/issues/1750 to create such tags
/* eslint "jsdoc/check-tag-names": ["error", {definedTags: ["cli-arg"]}] */
/**
 * @file A CLI reporter against after-the-fact compiled (merged Mochawesome)
 * Mocha results.
 * @cli-arg {"doc"|"dot"|"json-stream"|"json"|"landing"|"list"|
 * "markdown"|"min"|"nyan"|"progress"|"spec"|"tap"|
 * "xunit"} [0="spec"] The Mocha reporter to use
 */

'use strict';

const reporterFile = process.argv[2]
  ? process.argv[2].replace(/^--/u, '')
  : 'spec';

// eslint-disable-next-line import/no-dynamic-require
const MochaReporter = require(`mocha/lib/reporters/${reporterFile}.js`);
const Suite = require('mocha/lib/suite.js');
const Test = require('mocha/lib/test.js');
const Runner = require('mocha/lib/runner.js');

const {constants: {
  EVENT_RUN_BEGIN,
  EVENT_RUN_END,
  EVENT_SUITE_BEGIN,
  EVENT_SUITE_END,
  EVENT_TEST_FAIL,
  EVENT_TEST_PASS,
  EVENT_TEST_PENDING
}} = require('mocha/lib/runner.js');

const {results, stats} = require('../mochawesome.json');

const runner = new Runner(
  new Suite('', null, true)
);
runner.stats = stats;

console.log('Mocha results:');

// eslint-disable-next-line no-new
new MochaReporter(runner);

runner.emit(EVENT_RUN_BEGIN);
results.forEach(({suites}) => {
  suites.forEach(function handleSuite (st) {
    const ste = Object.assign(new Suite(''), st);

    ste.suites.forEach((s, i) => {
      ste.suites[i] = handleSuite(s);
    });

    runner.emit(EVENT_SUITE_BEGIN, ste);
    ste.tests.forEach((ts) => {
      const tst = new Test('', () => {
        //
      });
      Object.entries(ts).forEach(([k, v]) => {
        // `fullTitle` is a string in mochawesome but a function in Mocha
        if (k !== 'fullTitle') {
          tst[k] = v;
        }
      });
      tst.parent = ste; // Seems to work
      const ev = tst.pass
        ? EVENT_TEST_PASS
        : tst.fail
          ? EVENT_TEST_FAIL
          // No distinct event for pending vs. skipped?
          : tst.pending
            ? EVENT_TEST_PENDING
            : tst.skipped
              ? EVENT_TEST_PENDING
              : null;
      if (!ev) {
        throw new Error(
          'Unexpected test event (not passing, failing, or pending): ' +
          tst.title
        );
      }

      runner.emit(ev, tst, tst.fail ? tst.err : undefined);
    });
    runner.emit(EVENT_SUITE_END, ste);

    return ste;
  });
});
runner.emit(EVENT_RUN_END);

const {
  // passes, tests, pending,
  end
} = stats;

/*
console.log(
  `Passing ${passes}/${tests}${pending ? `Pending ${pending}` : ''}`
);
*/
const endDate = new Date(Date.parse(end));
const lastRan = new Intl.DateTimeFormat('en-US', {
  year: 'numeric', month: 'numeric', day: 'numeric',
  hour: 'numeric', minute: 'numeric'
}).format(endDate);

console.log(
  `Tests finished: ${lastRan}`
);
```

**Comment by jennifer-shehane** [comment]
@brettz9 Since this is a very visual request. Can you take some screenshots of the current output and highlight which areas are 'noisy', which areas you want changes, etc? It's hard to tell exactly which parts of the CLI output you're wanting updated without this. Thanks!

**Comment by brettz9** [comment]
I'm not so much looking to get rid of the noise (I figure you have a way to suppress some of this). At this point, I'd like to keep the noise so I can better track what was going on, e.g.:

![image](https://user-images.githubusercontent.com/20234/75515683-994dca00-5a35-11ea-9806-1b2d72e93be1.png)

...or such as this test summary (but which doesn't have the individual results):

![image](https://user-images.githubusercontent.com/20234/75515714-b1254e00-5a35-11ea-9e09-9775b9ae69f7.png)

...or noise from other processes run in parallel, such as this Mongodb output (which I imagine I could also suppress) interspersing the test results:

![image](https://user-images.githubusercontent.com/20234/75515768-ce5a1c80-5a35-11ea-8a22-a475d58ea7eb.png)

...or noise from our coverage run during the process:

![image](https://user-images.githubusercontent.com/20234/75515911-2c86ff80-5a36-11ea-856b-f01c4fefccc7.png)

After all the UI tests run, I am just happy (as my provided code allows) to get this output again at the end:

![image](https://user-images.githubusercontent.com/20234/75515975-4d4f5500-5a36-11ea-96b2-8ee6a35ea245.png)

Or to optionally get a briefer summary (shown here also with the after-the-fact report one can get with nyc by `nyc report`) (and/or to show only the failing tests at the end):

![image](https://user-images.githubusercontent.com/20234/75516417-6278b380-5a37-11ea-9358-94a2328052b6.png)

Even if I had no noise, it can be useful to have a `report` script to get back at and show the last aggregate test data results (e.g., if I closed a terminal window and wanted to get a quick look again at where our test results were last at).

**Comment by jennifer-shehane** [comment]
The following output is not output by Cypress, so is completely outside of our control. You'll have to suppress these logs yourself if you do not want them:

- Mongodb output
- coverage run report

Ultimately this looks like a duplicate of #2071

**Comment by brettz9** [comment]
I am aware these are not created by Cypress. I was just offering code in case it might be useful in your constellation of tools. Others may have noisy logs and rather than disabling those logs which can be useful to examine, be able to get a summary after the fact which collects the test results only.
